File: BassBlog/Classes/Controllers/Root/TabBar/AllMixes/View/BBAllMixesTableViewCell.m
//
//  BBAllMixesTableViewCell.m
//  BassBlog
//
//  Created by Evgeny Sivko on 26.02.14.
//  Copyright (c) 2014 BassBlog. All rights reserved.
//

#import "BBAllMixesTableViewCell.h"
#import "MarqueeLabel.h"

@interface BBAllMixesTableViewCell()

@property (nonatomic, strong) IBOutlet MarqueeLabel *label;

@end

@implementation BBAllMixesTableViewCell

@dynamic label;

- (void)awakeFromNib
{
    [super awakeFromNib];
    
    self.label.trailingBuffer = 30.f;
    self.label.animationDelay = 1.0;
    self.label.rate = 30.f;
    self.label.textAlignment = NSTextAlignmentLeft;
    self.label.marqueeType = MLContinuous;
}

- (void)setPaused:(BOOL)paused
{
    [super setPaused:paused];

    self.label.labelize = paused;
}

@end

File: BassBlog/Classes/Controllers/Root/TabBar/AllMixes/View/BBAllMixesTableViewCell.swift
//
//  BBAllMixesTableViewCell.swift
//  BassBlog
//
//  Created by M Ivaniushchenko on 6/9/19.
//  Copyright © 2019 BassBlog. All rights reserved.
//

import Foundation
import MarqueeLabel

@objc(BBAllMixesTableViewCell)
class BBAllMixesTableViewCell: BBMixesTableViewCell {
    @IBOutlet private var scrollingLabel: MarqueeLabel!
    
    override var label: UILabel! {
        get { return self.scrollingLabel }
        set {}
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        self.scrollingLabel.trailingBuffer = 30.0
        self.scrollingLabel.animationDelay = 1.0
        self.scrollingLabel.speed = .rate(30.0)
        self.scrollingLabel.textAlignment = .left
        self.scrollingLabel.type = .continuous
        self.scrollingLabel.holdScrolling = false
    }
    
    override var paused: Bool {
        didSet {
            self.scrollingLabel.labelize = self.paused
        }
    }
}
